amiwrap.c

```diff
@@ -72,348 +71,15 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
-                        
-                        if(ne>0){
-                            if(nmaxevent>0){
-                                if (tnext == discs[iroot]) {
-/* Possibly some change here ... */
-                                    handleEventB(&status, iroot, ami_mem, udata, tdata);
-                                    iroot--;
-                                }
-                            }
-                        }
-                        
-                        /* handle data-point */
-                        
-                        if (tnext == ts[it]) {
-/* Changes inside or outside? */
-                            handleDataPointB(&status, it, ami_mem, udata, rdata, tdata);
-                            it--;
-                        }
-                        
-                        /* reinit states */
-                        status = AMIReInitB(ami_mem, which, t, xB, dxB);
-                        if (status != AMI_SUCCESS) goto freturn;
-/* Maybe, but probably not, this needs to be changed */
-                        status = AMIQuadReInitB(ami_mem, which, xQB);
-                        if (status != AMI_SUCCESS) goto freturn;
-                        
-                        status = AMICalcICB(ami_mem, which, t, xB, dxB);
-                        if (status != AMI_SUCCESS) goto freturn;
-                    }
-                    
-                    /* we still need to integrate from first datapoint to tstart */
-                    if (t>tstart) {
-                        if(status == 0) {
-                            if (nx>0) {
-                                /* solve for backward problems */
-                                status = AMISolveB(ami_mem, tstart, AMI_NORMAL);
-                                if (status != AMI_SUCCESS) goto freturn;
-                                
-/* Here again, Quadrature needs a slight change, but probably inside... */
-                                status = AMIGetQuadB(ami_mem, which, &t, xQB);
-                                if (status != AMI_SUCCESS) goto freturn;
-                                status = AMIGetB(ami_mem, which, &t, xB, dxB);
-                                if (status != AMI_SUCCESS) goto freturn;
-                            }
-                        }
-                    }
-                    
-                    /* evaluate initial values */
-                    NVsx = N_VCloneVectorArray_Serial(np,x);
-                    if (NVsx == NULL) goto freturn;
-                    
-                    status = fx0(x,udata);
-                    if (status != AMI_SUCCESS) goto freturn;
-                    status = fdx0(x,dx,udata);
-                    if (status != AMI_SUCCESS) goto freturn;
-                    status = fsx0(NVsx, x, dx, udata);
-                    if (status != AMI_SUCCESS) goto freturn;
-                    
-                    if(status == 0) {
+    booleantype setupBdone = false;
```

Comment:
Variable definitions should be done before any code is called, otherwise we do not fullfill the C89 standard and AMICI cannot be compiled on VS2014<.